src/test/resources/refaster/RefasterAnyOfRecipes.java

```diff
@@ -190,6 +196,30 @@ public J visitMethodInvocation(J.MethodInvocation elem, ExecutionContext ctx) {
                     return super.visitMethodInvocation(elem, ctx);
                 }
 
+                @Override
+                public J visitNewClass(J.NewClass elem, ExecutionContext ctx) {
+                    JavaTemplate.Matcher matcher;
+                    if ((matcher = before$0.matcher(getCursor())).find()) {
+                        maybeRemoveImport("java.util.LinkedList");
+                        return embed(
+                                after.apply(getCursor(), elem.getCoordinates().replace()),
+                                getCursor(),
+                                ctx,
+                                SHORTEN_NAMES
+                        );
+                    }
+                    if ((matcher = before$1.matcher(getCursor())).find()) {
```

Comment:
In a perfect world we'd only match `before$0` in `visitNewClass`,
and only match `before$1` in `visitMethodInvocation`.